pkg/client/client.go

```diff
@@ -71,6 +71,14 @@ type CacheOptions struct {
 // NewClientFunc allows a user to define how to create a client.
 type NewClientFunc func(config *rest.Config, options Options) (Client, error)
 
+// NewAPIReaderFunc allows a user to define how to create an API server reader.
+type NewAPIReaderFunc func(config *rest.Config, options Options) (Reader, error)
+
+// NewAPIReader creates a new API server reader.
+func NewAPIReader(config *rest.Config, options Options) (Reader, error) {
+	return New(config, options)
+}
+
```

Comment:
in this file the last changes are missing, compare `KcpClusterDiscoveryCacheSize` and `MapperWithContext`.

Reply:
https://github.com/kcp-dev/controller-runtime/pull/57/files

Reply:
Weird, I wonder how I lost that 👀